http-core/src/main/scala/org/apache/pekko/http/impl/engine/http2/Http2Blueprint.scala

```diff
@@ -198,6 +199,20 @@ private[http] object Http2Blueprint {
       Flow[FrameEvent].map(FrameRenderer.render).prepend(Source.single(Http2Protocol.ClientConnectionPreface)),
       Flow[ByteString].via(new Http2FrameParsing(shouldReadPreface = false, log)))
 
+  private def rapidResetMitigation(
+      settings: Http2ServerSettings): BidiFlow[FrameEvent, FrameEvent, FrameEvent, FrameEvent, NotUsed] = {
+    def frameCost(event: FrameEvent): Int = event match {
+      case _: FrameEvent.DataFrame         => 0
+      case _: FrameEvent.WindowUpdateFrame => 0 // TODO: should we throttle these?
+      case _                               => 1
+    }
+
+    BidiFlow.fromFlows(
+      Flow[FrameEvent],
+      Flow[FrameEvent].throttle(settings.resetsThrottleCost, settings.resetsThrottleInterval,
+        settings.resetsThrottleBurst, frameCost, ThrottleMode.Enforcing))
```

Comment:
1. As the default `throttle` implementation will do a scheduling, but we will always close/fails on the first rapid reset detected, maybe a simpler copy/impl can be better?

Reply:
Could you suggest an implementation?

Reply:
I think we can leverage the current TokenBucket, IIRC, the Netty fix  https://github.com/netty/netty/commit/58f75f665aa81a8cbcf6ffa74820042a285c5e61 does no scheduling.

Reply:
where is this TokenBucket? No class with that name in this repo. The Netty PR does not seem to have a TokenBucket either.

Reply:
`org.apache.pekko.util.TokenBucket` from `pekko-actor`

Reply:
The throttle code already uses a TokenBucket under the hood. See https://github.com/search?q=repo%3Aapache%2Fincubator-pekko%20tokenbucket&type=code

This issue is time sensitive. Pekko HTTP has no current release that has a solution for CVE-2023-44487

@jrudolph would you have any thoughts on this? Do you think the throttle adds excessive overhead?

Reply:
@He-Pin The throttle in 'Enforcing' mode like this PR uses does not seem to schedule - see this code in Throttle.scala.

```
      override def onPush(): Unit = {
        val elem = grab(in)
        val cost = costCalculation(elem)
        val delayNanos = tokenBucket.offer(cost)

        if (delayNanos == 0L) push(out, elem)
        else {
          if (mode eq Enforcing) failStage(new RateExceededException("Maximum throttle throughput exceeded."))
          else {
            currentElement = elem
            scheduleOnce(Throttle.TimerKey, delayNanos.nanos)
          }
        }
```

Reply:
Maybe @He-Pin is talking about the `scheduleOnce` here?

Reply:
The `scheduleOnce` in this code snippet does not happen because `mode eq Enforcing`.

Reply:
Thanks for clarifying, in this case it seems to be a false alarm